packages/sanity/src/core/comments/utils/transform-children/index.ts

```diff
@@ -0,0 +1,23 @@
+import {type ReactNode} from 'react'
+
+import {linkMiddleware} from './linkMiddleware'
+import {type Middleware} from './types'
+
+const middlewares: Middleware[] = [linkMiddleware]
+
+/**
+ * A function that transforms a ReactNode using a set of middlewares
+ */
+export function transformChildren(node: ReactNode): ReactNode {
+  if (!Array.isArray(node)) return node
+
+  return node.flatMap((item) => {
+    let transformedItem: ReactNode[] = [item]
+
+    for (const middleware of middlewares) {
+      transformedItem = transformedItem.flatMap(middleware)
+    }
```

Comment:
Forgive me, can you just give me a run down why we need to do this?

Reply:
We are iterating over middleware functions that transform the nodes and pass them on to the next function. Currently, we only have one function (`linkMiddleware`), but in the future, we might have other functions to perform further transformations. 

For example:

```tsx
const middlewares: Middleware[] = [linkMiddleware, studioLinksMiddleware]
```

It might be a bit overkill, but the intent was to keep `transformChildren` a bit generic and not having it only focus on the links.

Reply:
Makes sense, thank you :)